Feedback: ESLint ruleset

tylerbutler opened the pull request:
*I'm not sure how to best review this, so this PR is an attempt. If you have other ideas, they are welcome.*

I exported the complete config (using [eslint --print-config](https://eslint.org/docs/user-guide/command-line-interface#--print-config)) that applies to a TypeScript file in a project using our strict config. That was you can comment on specific rules. However, it's still tough to know what all the rules do. :( I don't have any good solutions, but below are links to the docs for the big rulesets/plugins we're using so rules are easier to look up.

- <https://eslint.org/docs/rules/>
- <https://typescript-eslint.io/rules/>
- <https://github.com/sindresorhus/eslint-plugin-unicorn/tree/v40.0.0/docs/rules>
- <https://github.com/import-js/eslint-plugin-import/tree/main/docs/rules>

There is also a second file, changes-from-minimal-to-strict.json, that is a diff of today's minimal config (used my 99% of packages) and the new strict config.

The following PRs contain examples of the practical effects of applying this config to some of our packages.

- [ ] #9478
- [ ] #9430
- [ ] #9444
- [ ] #9489

tylerbutler committed:
Export strict eslint ruleset for review

tylerbutler commented on common/build/eslint-config-fluid/strict-ts.json:
```diff
@@ -0,0 +1,1440 @@
+      "error"
+    ],
+    "@typescript-eslint/unbound-method": [
+      "off"
```
We want this enabled.

markfields replied:
Please close #9082 when this gets turned on. Thanks!

tylerbutler commented on common/build/eslint-config-fluid/strict-ts.json:
```diff
@@ -0,0 +1,1440 @@
+  ],
+  "reportUnusedDisableDirectives": true,
+  "rules": {
+    "@typescript-eslint/explicit-member-accessibility": [
+      "error",
+      {
+        "accessibility": "explicit",
+        "overrides": {
+          "accessors": "explicit",
+          "constructors": "no-public",
+          "methods": "explicit",
+          "properties": "explicit",
+          "parameterProperties": "explicit"
+        }
+      }
+    ],
```
This requires explicit accessibility for everything *except* constructors, which are public by default.

markfields replied:
Isn't everything public by default?  https://www.typescriptlang.org/docs/handbook/2/classes.html#public

tylerbutler replied:
Yes, sorry. What I meant was that people coming from languages other than JS find public by default to be very... unexpected. Requiring explicit declaration makes it much easier to reason about the code.

My rationale for constructors was that in most cases public  constructors are what devs expect. This is true in other languages as well. That said, when in doubt, consistency should lead the way, so let's change the constructors to be explicit too.

tylerbutler commented on common/build/eslint-config-fluid/strict-ts.json:
```diff
@@ -0,0 +1,1440 @@
+        }
+      }
+    ],
+    "@typescript-eslint/no-parameter-properties": [
+      "warn",
+      {
+        "allows": [
+          "private",
+          "private readonly",
+          "public readonly",
+          "readonly"
+        ]
+      }
+    ],
```
This prevents properties from being declared within the parameters of constructors unless they're private or readonly.

markfields replied:
This will probably be super disruptive to fix up, and I personally don't mind this TS language feature.  And what's the reasoning behind the specific restrictions here (only allow private/readonly)?

tylerbutler replied:
Declaring parameters in constructors is foreign to non-TS devs, but more importantly, they make it more difficult to find where class-level properties are declared when reading code. It's a variant of the "declare as close to use as possible" rule -- "declare all class members together."

But I agree that cleanup may be ugly. And it does require more code.

markfields replied:
> And it does require more code

At least the emitted JS is the same, it appears, based on a quick check in TS Playground.

If there were a rule to require each such param on its own line in the constructor declaration I'd definitely be in favor of that, I'm strict to do that myself.

tylerbutler commented on common/build/eslint-config-fluid/strict-ts.json:
```diff
@@ -0,0 +1,1440 @@
+    "linebreak-style": [
+      "off"
+    ],
+    "no-trailing-spaces": [
+      "off"
+    ],
```
Enable this.

tylerbutler commented on common/build/eslint-config-fluid/strict-ts.json:
```diff
@@ -0,0 +1,1440 @@
+    "@typescript-eslint/linebreak-style": [
+      "off"
+    ],
+    "@typescript-eslint/no-trailing-spaces": [
+      "off"
+    ],
```
Enable this.

tylerbutler commented on common/build/eslint-config-fluid/strict-ts.json:
```diff
@@ -0,0 +1,1440 @@
+    "unicorn/prefer-array-some": [
+      "error"
+    ],
+    "unicorn/prefer-at": [
+      "off"
+    ],
```
Consider enabling this: <https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-at.md>

tylerbutler commented on common/build/eslint-config-fluid/strict-ts.json:
```diff
@@ -0,0 +1,1440 @@
+    "unicorn/throw-new-error": [
+      "error"
+    ],
+    "dot-notation": [
+      "off",
+      {
+        "allowPattern": "^_",
+        "allowKeywords": true
+      }
+    ],
```
Enable this.

tylerbutler commented on common/build/eslint-config-fluid/strict-ts.json:
```diff
@@ -0,0 +1,1440 @@
+        "allowKeywords": true
+      }
+    ],
+    "no-unused-expressions": [
+      "off"
+    ],
```
Make a warning.

markfields replied:
May be redundant with tsc enforcement.

tylerbutler replied:
I may be in the minority, but I would rather unused expressions be a lint-time error than a compilation error.

tylerbutler commented on common/build/eslint-config-fluid/strict-ts.json:
```diff
@@ -0,0 +1,1440 @@
+    "use-isnan": [
+      "error"
+    ],
+    "@typescript-eslint/no-unsafe-argument": [
+      "off"
+    ],
```
Enable this.

markfields commented on common/build/eslint-config-fluid/strict-ts.json:
```diff
@@ -0,0 +1,1440 @@
+    "@typescript-eslint/brace-style": [
+      "off"
+    ],
+    "@typescript-eslint/comma-dangle": [
+      "error",
+      {
+        "arrays": "always-multiline",
+        "enums": "always-multiline",
+        "exports": "always-multiline",
+        "functions": "always-multiline",
+        "generics": "never",
+        "imports": "always-multiline",
+        "objects": "always-multiline",
+        "tuples": "always-multiline"
+      }
+    ],
```
Yeah, agreed

tylerbutler commented on common/build/eslint-config-fluid/strict-ts.json:
```diff
@@ -0,0 +1,1440 @@
+    "@typescript-eslint/adjacent-overload-signatures": [
+      "warn"
+    ],
+    "@typescript-eslint/consistent-type-assertions": [
+      "warn",
+      {
+        "assertionStyle": "as",
+        "objectLiteralTypeAssertions": "never"
+      }
+    ],
```
Also consider enabling <https://typescript-eslint.io/rules/consistent-type-imports>

tylerbutler commented on common/build/eslint-config-fluid/strict-ts.json:
```diff
@@ -0,0 +1,1440 @@
+    "unicorn/prefer-negative-index": [
+      "error"
+    ],
+    "unicorn/prefer-node-protocol": [
+      "error"
+    ],
```
I'm planning to disable this, because the node protocol support was only added in 12.20/14.13.1, and earlier versions just blow up with the syntax.

tylerbutler replied:
I'll enable [import/no-nodejs-modules](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-nodejs-modules.md) as a replacement. Using node built-ins should always be explicit.

tylerbutler committed:
update base

tylerbutler committed:
update strict config

sonalivdeshpande committed:
policy check fix (#9522)

heliocliu committed:
update build-tools dependency version (#9527)

tylerbutler commented on common/build/eslint-config-fluid/strict-ts.json:
```diff
@@ -0,0 +1,1442 @@
+        "import/no-named-as-default": [
+            1
+        ],
+        "import/no-named-as-default-member": [
+            1
+        ],
```
Disable this per <https://typescript-eslint.io/docs/linting/troubleshooting/#eslint-plugin-import>

markfields committed:
Always log DeltaConnectionFailureToConnect (#9529)

When analyzing sessions that fail to connect, we don't have the luxury of the "MultipleDeltaConnectionFailures" telemetry event. We are already logging JoinSession_end over and over again (only once every 8 seconds!) so also logging "DeltaConnectionFailureToConnect" every time (instead of just once) feels ok.

This also adds additional context of which step failed (JoinSession, GetWebsocketToken, or initializing socket.io). And makes a small tweak to how JoinSession logs extra props to reduce extra columns in our tables.

Also fixes #9185

sonalivdeshpande committed:
fix container-utils version (#9531)

sonalivdeshpande committed:
[bump] package version to 0.58.3000 for development after client release (#9533)

znewton committed:
gitrest: persist pre-computed full summaries in storage (#9523)

tylerbutler committed:
Add missing types to build tools (#9525)

tylerbutler committed:
Fix missing null check in repo format check (#9530)

chensixx committed:
adding client type (#9532)

anthony-murphy committed:
Snapshot Tests: Strict Channels and Non-comparable Channels (#9441)

The pervious implementation of Unknown Channels had a major issue where we were losing data simply because the data was non-deterministic.

This change does a few things. It allows adding channel factories that are non-deterministic, like shared matrix, but enabled excluding them from comparison. This allows use to keep the data, which will still be tested in hydration.

Additionally, this adds a mode, strictChannels to the replayTool. This is only set in the snapshot tests, and will cause failure if a channel type is not found. This will prevent the addition of new snapshots to the snapshot tests which include channels we don't have explicit handling for. This will prevent further data loss.

TestData PR: microsoft/FluidFrameworkTestData#40

related #6957 #9405

hedasilv committed:
[GitRest] Adding isomorphic-git implementation of git functionality (#9537)

* Adding isomorphic-git to GitRest

* Adding isomorphic-git to GitRest

* Fixing lint errors

* Allowing isomorphic-git to be used from runnerFactory

* Updating use of fs to include fs.promises, since isomorphic-git expects a .promises when dealing with the fs object

* Adding comments

* addressing comments

anthony-murphy committed:
Port: MergeTree: Fix overlap clients in snapshotv1 (#9405)

This is a backcompat port of #9369

It fixes #9138 in a back compat way

hedasilv committed:
[GitRest] Refactoring GitRest code to consume gitrest-base as a package (#9550)

* [GitRest] Refactoring GitRest code to consume gitrest-base as a package

* Fixing repo policy check errors

* More policy-check fixes

sonalivdeshpande committed:
Skipping e2e tests failed (#9543)

hedasilv committed:
Pipeline changes to enable publishin GitRest npm packages (#9554)

pleath committed:
Account for gaps in the LocalReference array when packing segments (#9565)

When Zamboni combines TextSegment's, it needs to update LocalReference's that refer to the segments. It wasn't doing this correctly in the case where, say, a reference is found at position 0, then some number of segments without references were appended, then a segment containing a reference was appended. The text corresponding to the segments without references wasn't accounted in the final segment. A gap needs to be introduced in the LocalReference array to make the references correct.

jatgarg committed:
Revert Shared directory breaking changes from main (#9582)

chensixx committed:
Adding clientType to loader layer and passed to driver (#9455)

curtisman committed:
Add README documentation for @fluidframework/test-version-utils (#9578)

dependabot[bot] committed:
Bump minimist from 1.2.5 to 1.2.6 in /common/lib/container-definitions (#9591)

Bumps [minimist](https://github.com/substack/minimist) from 1.2.5 to 1.2.6.
- [Release notes](https://github.com/substack/minimist/releases)
- [Commits](https://github.com/substack/minimist/compare/1.2.5...1.2.6)

---
updated-dependencies:
- dependency-name: minimist
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump minimist from 1.2.5 to 1.2.6 in /common/lib/driver-definitions (#9590)

Bumps [minimist](https://github.com/substack/minimist) from 1.2.5 to 1.2.6.
- [Release notes](https://github.com/substack/minimist/releases)
- [Commits](https://github.com/substack/minimist/compare/1.2.5...1.2.6)

---
updated-dependencies:
- dependency-name: minimist
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump minimist from 1.2.5 to 1.2.6 in /common/lib/common-utils (#9593)

Bumps [minimist](https://github.com/substack/minimist) from 1.2.5 to 1.2.6.
- [Release notes](https://github.com/substack/minimist/releases)
- [Commits](https://github.com/substack/minimist/compare/1.2.5...1.2.6)

---
updated-dependencies:
- dependency-name: minimist
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump minimist from 1.2.5 to 1.2.6 in /tools/build-tools (#9596)

Bumps [minimist](https://github.com/substack/minimist) from 1.2.5 to 1.2.6.
- [Release notes](https://github.com/substack/minimist/releases)
- [Commits](https://github.com/substack/minimist/compare/1.2.5...1.2.6)

---
updated-dependencies:
- dependency-name: minimist
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

tylerbutler committed:
Remove some peer dependencies (#9563)

Using peer dependencies with our build tooling is unneeded because the tools are intended only for use in our repo, and peer dependencies are a hassle for downstream consumers.

Note that we have legitimate peer dependencies declared in the azure-client and Property DDS packages. Those dependencies are not addressed in this PR. See #7913 for more details.

dependabot[bot] committed:
Bump minimist from 1.2.5 to 1.2.6 in /common/lib/common-definitions (#9588)

Bumps [minimist](https://github.com/substack/minimist) from 1.2.5 to 1.2.6.
- [Release notes](https://github.com/substack/minimist/releases)
- [Commits](https://github.com/substack/minimist/compare/1.2.5...1.2.6)

---
updated-dependencies:
- dependency-name: minimist
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump minimist from 1.2.5 to 1.2.6 in /common/lib/protocol-definitions (#9589)

Bumps [minimist](https://github.com/substack/minimist) from 1.2.5 to 1.2.6.
- [Release notes](https://github.com/substack/minimist/releases)
- [Commits](https://github.com/substack/minimist/compare/1.2.5...1.2.6)

---
updated-dependencies:
- dependency-name: minimist
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump minimist from 1.2.5 to 1.2.6 in /common/build/eslint-config-fluid (#9586)

Bumps [minimist](https://github.com/substack/minimist) from 1.2.5 to 1.2.6.
- [Release notes](https://github.com/substack/minimist/releases)
- [Commits](https://github.com/substack/minimist/compare/1.2.5...1.2.6)

---
updated-dependencies:
- dependency-name: minimist
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump minimist from 1.2.5 to 1.2.6 in /common/lib/core-interfaces (#9587)

Bumps [minimist](https://github.com/substack/minimist) from 1.2.5 to 1.2.6.
- [Release notes](https://github.com/substack/minimist/releases)
- [Commits](https://github.com/substack/minimist/compare/1.2.5...1.2.6)

---
updated-dependencies:
- dependency-name: minimist
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump minimist from 1.2.5 to 1.2.6 in /tools/generator-fluid (#9595)

Bumps [minimist](https://github.com/substack/minimist) from 1.2.5 to 1.2.6.
- [Release notes](https://github.com/substack/minimist/releases)
- [Commits](https://github.com/substack/minimist/compare/1.2.5...1.2.6)

---
updated-dependencies:
- dependency-name: minimist
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump minimist from 1.2.5 to 1.2.6 in /tools/benchmark (#9597)

Bumps [minimist](https://github.com/substack/minimist) from 1.2.5 to 1.2.6.
- [Release notes](https://github.com/substack/minimist/releases)
- [Commits](https://github.com/substack/minimist/compare/1.2.5...1.2.6)

---
updated-dependencies:
- dependency-name: minimist
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump minimist from 1.2.5 to 1.2.6 in /docs (#9600)

Bumps [minimist](https://github.com/substack/minimist) from 1.2.5 to 1.2.6.
- [Release notes](https://github.com/substack/minimist/releases)
- [Commits](https://github.com/substack/minimist/compare/1.2.5...1.2.6)

---
updated-dependencies:
- dependency-name: minimist
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

anthony-murphy committed:
Main: Fix type validation (#9594)

dependabot[bot] committed:
Bump minimist from 1.2.5 to 1.2.6 in /server/historian (#9598)

Bumps [minimist](https://github.com/substack/minimist) from 1.2.5 to 1.2.6.
- [Release notes](https://github.com/substack/minimist/releases)
- [Commits](https://github.com/substack/minimist/compare/1.2.5...1.2.6)

---
updated-dependencies:
- dependency-name: minimist
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump minimist from 1.2.5 to 1.2.6 in /server/routerlicious (#9599)

Bumps [minimist](https://github.com/substack/minimist) from 1.2.5 to 1.2.6.
- [Release notes](https://github.com/substack/minimist/releases)
- [Commits](https://github.com/substack/minimist/compare/1.2.5...1.2.6)

---
updated-dependencies:
- dependency-name: minimist
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump minimist from 1.2.5 to 1.2.6 in /server/gitrest (#9601)

Bumps [minimist](https://github.com/substack/minimist) from 1.2.5 to 1.2.6.
- [Release notes](https://github.com/substack/minimist/releases)
- [Commits](https://github.com/substack/minimist/compare/1.2.5...1.2.6)

---
updated-dependencies:
- dependency-name: minimist
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

markfields committed:
Update labeler.yml to use updated "area: tests" label (#9604)

heliocliu committed:
Add labeling for branches (#9602)

Add labeling for branches using srvaroa/labeler and move the config file for the existing actions/labeler since srvaroa doesn't seem to have that option

anthony-murphy committed:
Main: Delete Unecessary Package Lock Files and Fix Driver Defs Version (#9606)

heliocliu committed:
Fix branch labeler config (#9612)

tylerbutler committed:
eslint config: Fix modern module resolution (#9562)

tylerbutler commented on common/build/eslint-config-fluid/strict-ts.json:
```diff
@@ -0,0 +1,1442 @@
+                "allowTypedFunctionExpressions": true
+            }
+        ],
+        "@typescript-eslint/explicit-member-accessibility": [
+            "error",
+            {
+                "accessibility": "explicit",
+                "overrides": {
+                    "accessors": "explicit",
+                    "constructors": "no-public",
+                    "methods": "explicit",
+                    "parameterProperties": "explicit",
+                    "properties": "explicit"
+                }
+            }
+        ],
```
set constructors to explicit as well

tylerbutler commented on common/build/eslint-config-fluid/strict-ts.json:
```diff
@@ -0,0 +1,1442 @@
+        "@typescript-eslint/eol-last": [
+            "off"
+        ],
+        "@typescript-eslint/explicit-function-return-type": [
+            "error",
+            {
+                "allowConciseArrowFunctionExpressionsStartingWithVoid": false,
+                "allowDirectConstAssertionInArrowFunctions": true,
+                "allowExpressions": false,
+                "allowHigherOrderFunctions": true,
+                "allowTypedFunctionExpressions": true
+            }
+        ],
```
Set allowExpressions to true.

NicholasCouri committed:
Adding nonSysOps count to the treeSnapshot (#9622)

hedasilv committed:
[t9s] Avoiding setting up Lumberjack twice (#9634)

sonalivdeshpande committed:
move raiseContainerWarning to 0.58 upcoming change (#9636)

tylerbutler committed:
Add missing step in readme instructions (#9625)

tyler-cai-microsoft committed:
Log gc blob size (#9485)

* Log gc blob size

NicholasCouri committed:
Initial draft for collecting Inbound Queue wait time  (#9504)

*Collecting Inbound Queue wait.

sonalivdeshpande committed:
Remove raiseContainerWarning from IContainerContext (#9654)

markfields committed:
Ensure all packages depend on common-definitions 20.1 (#8624)

Months ago I made the mistake of introducing a breaking change between 20.0 and 20.1, and client consumes 0.20.1. So it's best to get everyone on the same page here. Anyway, due to semver and npm install behavior, I'm pretty certain that anyone consuming client will be installing 0.20.1 so this shouldn't cause any changes as it rolls out.

Specifically, this causes breaks in some cases when trying to do --symlink:full.

dependabot[bot] committed:
Bump ansi-regex from 4.1.0 to 4.1.1 in /common/lib/common-definitions (#9665)

Bumps [ansi-regex](https://github.com/chalk/ansi-regex) from 4.1.0 to 4.1.1.
- [Release notes](https://github.com/chalk/ansi-regex/releases)
- [Commits](https://github.com/chalk/ansi-regex/compare/v4.1.0...v4.1.1)

---
updated-dependencies:
- dependency-name: ansi-regex
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump ansi-regex from 5.0.0 to 5.0.1 in /common/lib/protocol-definitions (#9663)

Bumps [ansi-regex](https://github.com/chalk/ansi-regex) from 5.0.0 to 5.0.1.
- [Release notes](https://github.com/chalk/ansi-regex/releases)
- [Commits](https://github.com/chalk/ansi-regex/compare/v5.0.0...v5.0.1)

---
updated-dependencies:
- dependency-name: ansi-regex
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump ansi-regex from 3.0.0 to 3.0.1 in /tools/benchmark (#9668)

Bumps [ansi-regex](https://github.com/chalk/ansi-regex) from 3.0.0 to 3.0.1.
- [Release notes](https://github.com/chalk/ansi-regex/releases)
- [Commits](https://github.com/chalk/ansi-regex/compare/v3.0.0...v3.0.1)

---
updated-dependencies:
- dependency-name: ansi-regex
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump ansi-regex from 5.0.0 to 5.0.1 in /common/lib/driver-definitions (#9669)

Bumps [ansi-regex](https://github.com/chalk/ansi-regex) from 5.0.0 to 5.0.1.
- [Release notes](https://github.com/chalk/ansi-regex/releases)
- [Commits](https://github.com/chalk/ansi-regex/compare/v5.0.0...v5.0.1)

---
updated-dependencies:
- dependency-name: ansi-regex
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump ansi-regex from 3.0.0 to 3.0.1 in /tools/build-tools (#9666)

Bumps [ansi-regex](https://github.com/chalk/ansi-regex) from 3.0.0 to 3.0.1.
- [Release notes](https://github.com/chalk/ansi-regex/releases)
- [Commits](https://github.com/chalk/ansi-regex/compare/v3.0.0...v3.0.1)

---
updated-dependencies:
- dependency-name: ansi-regex
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump ansi-regex from 3.0.0 to 3.0.1 in /tools/generator-fluid (#9664)

Bumps [ansi-regex](https://github.com/chalk/ansi-regex) from 3.0.0 to 3.0.1.
- [Release notes](https://github.com/chalk/ansi-regex/releases)
- [Commits](https://github.com/chalk/ansi-regex/compare/v3.0.0...v3.0.1)

---
updated-dependencies:
- dependency-name: ansi-regex
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump ansi-regex from 3.0.0 to 3.0.1 in /common/lib/common-utils (#9667)

Bumps [ansi-regex](https://github.com/chalk/ansi-regex) from 3.0.0 to 3.0.1.
- [Release notes](https://github.com/chalk/ansi-regex/releases)
- [Commits](https://github.com/chalk/ansi-regex/compare/v3.0.0...v3.0.1)

---
updated-dependencies:
- dependency-name: ansi-regex
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

jatgarg committed:
Revert the revert of Shared Directory changes and make them non breaking in main (#9659)

ChumpChief committed:
Split view/model for simple-fluidobject-embed and Pond examples (#9616)

sonalivdeshpande committed:
Adding raiseContainerWarning in main (#9675)

anthony-murphy committed:
Type Validation Tool Improvements (#9674)

Make core type validation logic async to support parallelism on io, and make main loop parrallel
Improve error handling in core type validation logic
Enable loading from packages without source (from .d.ts) as we strip source from some packages, like aqueduct
Improve type handling, specifically add TypeOnly utility the recursively turns all types into typescript types, otherwise we hit problems with privates on classes.
Switch to single back compat test (Previous). The old format was overkill. In this new way every version will validation only it's previous, which will create a chain of validations between all versions. We have functional e2e test that validate back compat requirements which are runtime, not compile time.
To see changes in action look at #9673

jatgarg committed:
Add stress test for testing new fluid binary wire format (#9391)

jatgarg committed:
Disbale batching test (#9678)

chensixx committed:
Fixing runWithRetry with abort signal when container closes (#9195)

ChumpChief committed:
Reenable unbound method lint rule and preemptive fixes (#9689)

hedasilv committed:
[GitRest] Refactoring APIs to support storage name header (#9672)

* [GitRest] Refactoring APIs to support storage name header

* Addressing comments

hedasilv committed:
[r11s] Allowing getOrCreateRepo to handle headers (#9691)

anthony-murphy committed:
Enable type validation for all client packages (#9673)

This change uses a new improved version of the type validation tool to generate validation tests for all packages (#9674).

Generally this change in all done via automation, but there are a couple other changes.

Remove type validation from existing packages, and move to new model
Fix fluid-build non-conformant packages
All automatic type validation generation is removed. It is to slow to run on every build. It should only need to be run after release, and in the presence of new breaking changes that must be suppressed

jatgarg committed:
Add odsp driver support to upload container type summary (#9671)

jatgarg committed:
Add breaking.md (#9701)

ChumpChief committed:
Integrate latest eslint-config-fluid prerelease (#9696)

anthony-murphy committed:
Update build tools and Update Common Libs to use new Type Generation (#9698)

This change updates the build tools, and integrates the changes to type generation in the new version of the tools. See #9674

sonalivdeshpande committed:
Update breaking.md with raiseContainerWarning removal (#9705)

ChumpChief committed:
Convert HashCalculator to .ts + class and use standard lint rules for property-inspector-table (#9624)

* Convert HashCalculator to TS
* Convert HashCalculator to es6 class
* Update lint rules to scan all src except *.stories.* (which are excluded from tsconfig)
* Update lint rules to be our normal set, minus some exceptions that require more involved fixes
* Fix lint errors that were relatively lower-effort

andre4i committed:
Track usages of `setFlushMode` and announce its deprecation (#9695)

First step in making flushMode immutable for a container

Abe27342 committed:
Update generated report for @fluid-experimental/tree (#9713)

Co-authored-by: Abram Sanderson <absander@microsoft.com>

vladsud committed:
Add better tracking of Collab window (#9707)

Closes #9683: Duration in "MsnStatistics" event is tracking garbage
Helps better understand #9682: MSN window is too large!

This change adds
- tracking of collab window at summarization, to better understand impact on quality of summary (search, size of catchup blobs).
- tracking of batch sizes, time to inbound and msn size at that time
   - I expect to see msn size correlating with batch sizes, as collab window grows to at least the size of batch at the moment when batch is sent
- Correct tracking of how long it takes for a sequence number to fall out of collab window.

wes-carlson committed:
correctly handle simultaneous upload of identical blob (#9716)

Correctly handle the case where multiple containers upload the same blob and submit blobAttach ops with identical blob IDs due to blob de-duping. In this case, BlobManager should only resolve the deferred promise when it sees its own op.

tylerbutler commented:
This is now covered by #9740 and #9748, so I'm closing this.
